File: modern_android/room_exam/app/src/main/java/com/jsoh/room_exam/MainActivity.java
package com.jsoh.room_exam;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.lifecycle.ViewModelProvider;

import com.jsoh.room_exam.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setLifecycleOwner(this);

        MainViewModel viewModel = new ViewModelProvider(this).get(MainViewModel.class);
        binding.setViewModel(viewModel);

        // UI 갱신
        /*
        viewModel.getAll().observe(this, todos -> {
            binding.resultText.setText(todos.toString());
        });
        //*/

        /*
        // 버튼 클릭시 DB에 insert
        findViewById(R.id.add_button).setOnClickListener(v -> {
            viewModel.insert(binding.todoEdit.getText().toString());
        });
        //*/
    }
}

File: modern_android/room_exam_kotlin/app/src/main/java/com/jsoh/room_exam_kotlin/MainActivity.kt
package com.jsoh.room_exam_kotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import com.jsoh.room_exam_kotlin.databinding.ActivityMainBinding
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        // LiveData 때문에 추가되는 코드임
        binding.lifecycleOwner = this

        val viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        binding.viewModel = viewModel

        /*
        viewModel.getAll().observe(this, Observer {
            result_text.text = it.toString()
        })
        */

        /*
        add_button.setOnClickListener {
            lifecycleScope.launch(Dispatchers.IO) {
                viewModel.insert(todo_edit.text.toString())
            }
        }
        */
    }
}